packages/next-app/src/app/internalmanager/(components)/TokenTable.tsx

```diff
@@ -0,0 +1,133 @@
+"use client";
+import { InternalBalanceQuery } from "@balancer-pool-metadata/gql/src/balancer-internal-manager/__generated__/Mainnet";
+import { networkFor } from "@balancer-pool-metadata/shared";
+import Image from "next/image";
+import Link from "next/link";
+import { tokenLogoUri } from "public/tokens/logoUri";
+import { useAccount, useNetwork } from "wagmi";
+
+import { ToastContent } from "#/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/TransactionModal";
+import genericTokenLogo from "#/assets/generic-token-logo.png";
+import { Button } from "#/components";
+import Table from "#/components/Table";
+import { Toast } from "#/components/Toast";
+import { useInternalBalance } from "#/contexts/InternalManagerContext";
+import { impersonateWhetherDAO, internalBalances } from "#/lib/gql";
+import { ArrElement, GetDeepProp } from "#/utils/getTypes";
+
+export function TokenTable() {
+  const { chain } = useNetwork();
+  let { address } = useAccount();
+  address = impersonateWhetherDAO(chain?.id.toString() || "1", address);
+
+  const addressLower = address ? address?.toLowerCase() : "";
+
+  const { notification, setIsNotifierOpen, isNotifierOpen, transactionUrl } =
+    useInternalBalance();
+
+  const { data: internalBalanceData } = internalBalances
+    .gql(chain?.id.toString() || "1")
+    .useInternalBalance({
+      userAddress: addressLower as `0x${string}`,
+    });
+
+  const tokensWithBalance =
```

Comment:
we could do this at the query-level:

`internalBalances(where: { balance_gt: 0 }) { ... }`